lib/runners/harness.rb

```diff
@@ -91,5 +94,35 @@ def handle_error(exn)
         trace_writer.error "#{exn.message} (#{exn.class})"
       end
     end
+
+    def remove_ignored_files(config)
+      ignores = config.ignore
+      return if ignores.empty?
+
+      trace_writer.message("Deleting ignored files...") do
+        with_gitignore(ignores) do |list|
+          FileUtils.rm_rf(list)
+          trace_writer.message list.map { |f| f.relative_path_from(working_dir) }.join("\n")
+        end
+      end
+    end
+
+    def with_gitignore(ignores)
+      gitignore = (working_dir / ".gitignore")
+
+      backup = gitignore.file? ? gitignore.read : nil
+      gitignore.write(ignores.join("\n"))
+
+      shell = Shell.new(current_dir: working_dir, trace_writer: trace_writer, env_hash: {})
+      shell.capture3!("git", "init", trace_command_line: false, trace_stdout: false)
+
+      # @see https://git-scm.com/docs/git-check-ignore#_exit_status
+      stdout, = shell.capture3!("git", "check-ignore", *working_dir.glob("**/*", File::FNM_DOTMATCH).map(&:to_s), trace_command_line: false, trace_stdout: false, is_success: -> (s) { s.exitstatus != 128 })
```

Comment:
[suggestion] Can you format the code layout like this? I feel too long line hard to read... 😢 

```suggestion
      stdout, = shell.capture3!(
        "git", "check-ignore", *working_dir.glob("**/*", File::FNM_DOTMATCH).map(&:to_s),
        trace_command_line: false, trace_stdout: false, is_success: -> (s) { s.exitstatus != 128 })
```